pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -823,5 +823,12 @@ self: super: builtins.intersectAttrs super {
         ln -s $out/bin/haskell-language-server $out/bin/haskell-language-server-${ghc_version}
         ln -s $out/bin/haskell-language-server $out/bin/haskell-language-server-${ghc_major_version}
        '';
+    testToolDepends = [ self.cabal-install pkgs.git ];
+    testTarget = "func-test"; # wrapper test accesses internet
+    preCheck = ''
+      export PATH=$PATH:$(pwd)/dist/build/haskell-language-server:$(pwd)/dist/build/haskell-language-server-wrapper
+      mkdir home
+      export HOME=$(pwd)/home
```

Comment:
In nixpkgs you often see `export HOME=$TMPDIR` (but there does seem to be a lot of variation in this).

Reply:
Well, it's a function call less, so I'll go with that.